dataplane/src/pipeline/test_utils.rs

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Open Network Fabric Authors
+
+#![allow(clippy::unwrap_used, clippy::expect_used, clippy::panic)]
+
+use net::buffer::TestBuffer;
+use net::eth::Eth;
+use net::eth::ethtype::EthType;
+use net::eth::mac::{DestinationMac, Mac, SourceMac};
+use net::headers::{Headers, Net};
+use net::ipv4::Ipv4;
+use net::ipv4::addr::UnicastIpv4Addr;
+use net::parse::DeParse;
+use std::default::Default;
+use std::net::Ipv4Addr;
+
+use crate::packet::{InvalidPacket, Packet};
+use crate::pipeline::sample_nfs::{BroadcastMacs, DecrementTtl, InspectHeaders, Passthrough};
+use crate::pipeline::{DynNetworkFunction, nf_dyn};
+
+/// Generates an infinite sequence of network functions.
+///
+/// The sequence is a repeating pattern of:
+/// - [`InspectHeaders`]
```

Comment:
nit: this is another place where I would suggest putting the docs on the function rather than the type

Reply:
Fixed.